Javatrix.java

```diff
@@ -477,5 +477,75 @@ public Javatrix times(double scalar)
         }
         return actual;
     }
+
+    /**
+     * Get a submatrix.
+     *
+     * @param r - an array of row indices
+     * @param c - an array of column indices
+     * @return submatrix(r(:), c(:))
+     * @throws ArrayIndexOutOfBoundsException if submatrix
+     * has larger dimensions than this.matrix.
+     */
+    public Javatrix  getMatrix(int[] r, int[] c)
+        throws ArrayIndexOutOfBoundsException
+    {
+        m = r.length;
+        n = c.length;
+        int row;
+        int col;
+        if (m > this.getRowDimension() || n > this.getColumnDimension())
+        {
+            String ex = "Submatrix must have dimensions "
+                + "less than or equal to matrix.";
+            throw new ArrayIndexOutOfBoundsException(ex);
+        }
+        Javatrix subMatrix = new Javatrix(m, n);;
+        for (int i = 0; i < m; i++)
+        {
+            row = r[i];
+            col = c[i];
+            subMatrix.matrix[row][col] = this.get(row, col);
+        }
+        return subMatrix;
+    }
+
+
+    /**
+     * Get a submatrix.
+     *
+     * @param r - an array of row indices
+     * @param j0 - the initial column index
+     * @param j1 - the final column index
+     * @return submatrix(r(:), j0:j1)
+     * @throws ArrayIndexOutOfBoundsException if submatrix
+     * has larger dimensions than this.matrix.
+     */
+    public Javatrix getMatrix(int[] r, int j0, int j1)
+        throws ArrayIndexOutOfBoundsException
+    {
+        m = r.length;
+        n = j1 - j0 + 1;
+        int row;
+        int col = 0;
+        if (m > this.getRowDimension() || j1 > this.getColumnDimension())
```

Comment:
See issue #23 review regarding exception test and r and c equal length test. Also, both j0 and j1 need to be checked to make sure they fall within bounds of the calling Javatrix object's matrix.